src/Codeception/Module/Redis.php

```diff
@@ -3,87 +3,626 @@
 namespace Codeception\Module;
 
 use Codeception\Module as CodeceptionModule;
+use Codeception\TestCase;
+use Codeception\Exception\ModuleException;
 use Codeception\TestInterface;
-use Codeception\Exception\ModuleException as ModuleException;
-use Codeception\Lib\Driver\Redis as RedisDriver;
+use Predis\Client as RedisDriver;
 
 /**
- * Works with Redis database.
- *
- * Cleans up Redis database after each run.
+ * This module uses the [Predis](https://github.com/nrk/predis) library
+ * to interact with a Redis server.
  *
  * ## Status
  *
- * * Maintainer: **judgedim**
  * * Stability: **beta**
- * * Contact: https://github.com/judgedim
  *
  * ## Configuration
  *
- * * host *required* - redis host to connect
- * * port *required* - redis port.
- * * database *required* - redis database.
- * * cleanup: true - defined data will be purged before running every test.
+ * * **`host`** (`string`, default `'127.0.0.1'`) - The Redis host
+ * * **`port`** (`int`, default `6379`) - The Redis port
+ * * **`database`** (`int`, no default) - The Redis database. Needs to be explicitely specified.
+ * * **`cleanupBefore`**: (`string`, default `'suite'`) - Whether/when to flush the database:
+ *     * `suite`: at the beginning of every suite
+ *     * `test`: at the beginning of every test
+ *     * Any other value: never
  *
  * ## Public Properties
- * * driver - contains Connection Driver
- *
- * ### Beta Version
+ * * **driver** - Contains the Predis client/driver
  *
- * Report an issue if this module doesn't work for you.
- *
- * @author judgedim
+ * @author Marc Verney <marc@marcverney.net>
  */
 class Redis extends CodeceptionModule
 {
+    /**
+     * {@inheritdoc}
+     *
+     * No default value is set for the database, as this module will delete
+     * every data in it. The user is required to explicitely set this parameter.
+     */
     protected $config = [
-        'cleanup' => true
```

Comment:
I'd rather keep this setting. I think redis should be cleaned up before all tests by default.